### PR TITLE
Revoke token when user logout.

### PR DESCRIPTION
**Pull request by corentindrouet**
in response of issue #35 

/oauth/revoke end point destroy the access/refresh token when logout.

### CODE REVIEW

File: config/oauth2.js [hunk 1]
```diff
@@ -170,6 +170,33 @@ module.exports = {
...
+        function(req, res) {
+          if (req.body.token_type_hint === 'access_token') {
+            AccessToken.destroy({
+              token: req.body.token
+            })
+              .then(() => {
+                return res.send(200);
+              })
+              .catch(() => {
+                return res.badRequest();
```
Comment (willrnch) [inline config/oauth2.js:183]: replace by: `.catch((err) => res.negotiate(err));`

File: config/oauth2.js [hunk 2]
```diff
@@ -170,6 +170,33 @@ module.exports = {
...
+              });
+          } else if (req.body.token_type_hint === 'refresh_token') {
+            RefreshToken.destroy({
+              token: req.body.token
+            })
+              .then(() => {
+                return res.send(200);
+              })
+              .catch(() => {
+                return res.badRequest();
```
Comment (willrnch) [inline config/oauth2.js:193]: replace by: .catch((err) => res.negotiate(err));

File: config/oauth2.js [hunk 3]
```diff
@@ -170,6 +170,33 @@ module.exports = {
...
+              });
+          } else {
+            return res.badRequest();
```
Comment (willrnch) [inline config/oauth2.js:196]: Issue an explicit error "Invalid to type hint".

File: tests/api/auth.test.js [hunk 4]
```diff
@@ -115,4 +116,97 @@ module.exports = function() {
...
+            throw new Error('Access Token should be deleted');
+          } else if (res.refresh.length !== 0) {
+            throw new Error('Refresh Token should be deleted');
+          } else {
+            User.destroy({
+              email: 'revoke@token.com'
+            })
+              .then(() => {
+                done();
+              });
```
Comment (willrnch) [inline tests/api/auth.test.js:207]: missing error handling

**Comment by Leblantoine** [comment]
Closed in favor of PR : #93 
The comments have been considered.

**Comment by Leblantoine** [comment]
Closed in favor of PR : #93 
The comments have been considered.

**Pull request closed without merging**